src/Show.js

```diff
@@ -3,13 +3,41 @@ import "./Show.css";
 
 const tumblr_re = /^\d+?\.media\.tumblr\.com$/;
 
+const getGfycatUrl = urlPathname => {
+  const gfycatId = urlPathname.substr(1);
+  const storageKey = "gfycatUrls";
+
+  return new Promise((resolve, reject) => {
+    let gfycatUrls;
+    try {
+      gfycatUrls = JSON.parse(sessionStorage.getItem(storageKey) || "{}");
+    } catch (e) {
+      gfycatUrls = {};
+    }
+    const entry = gfycatUrls[gfycatId];
+    if (entry && entry.expiresAt > new Date()) {
+      resolve(entry.url);
+      return;
+    }
+    fetch(`https://api.gfycat.com/v1/gfycats${urlPathname}`)
+      .then(r => r.json())
+      .then(data => {
+        const url = data.gfyItem.webmUrl;
+        const expiresAt = new Date().getTime() + 1 * 24 * 60 * 60 * 1000; // 1 day
+        gfycatUrls[gfycatId] = { url, expiresAt };
```

Comment:
not sure this makes much of a difference, but you can save characters in the serialization if instead of an object you made it an array. of course, this can result in a lack of clarity about what each index represents, so i will leave this judgment up to you

Reply:
I've done the other things, but saving characters here seems like a micro-optimization at the cost of readability. Storage space shouldn't be a problem now that expired entries will get removed.